src/apis/servicenetworking/v1.ts

```diff
@@ -0,0 +1,3036 @@
+   * not bound by the path template automatically become HTTP query parameters
+   * if there is no HTTP request body. For example:      service Messaging { rpc
+   * GetMessage(GetMessageRequest) returns (Message) {         option
+   * (google.api.http) = {             get:&quot;/v1/messages/{message_id}&quot;
+   * };       }     }     message GetMessageRequest {       message SubMessage {
+   * string subfield = 1;       }       string message_id = 1; // Mapped to URL
+   * path.       int64 revision = 2;    // Mapped to URL query parameter
+   * `revision`.       SubMessage sub = 3;    // Mapped to URL query parameter
+   * `sub.subfield`.     }  This enables a HTTP JSON to RPC mapping as below:
+   * HTTP | gRPC -----|----- `GET
+   * /v1/messages/123456?revision=2&amp;sub.subfield=foo` |
+   * `GetMessage(message_id: &quot;123456&quot; revision: 2 sub:
+   * SubMessage(subfield: &quot;foo&quot;))`  Note that fields which are mapped
+   * to URL query parameters must have a primitive type or a repeated primitive
+   * type or a non-repeated message type. In the case of a repeated type, the
+   * parameter can be repeated in the URL as `...?param=A&amp;param=B`. In the
+   * case of a message type, each field of the message is mapped to a separate
+   * parameter, such as `...?foo.a=A&amp;foo.b=B&amp;foo.c=C`.  For HTTP methods
+   * that allow a request body, the `body` field specifies the mapping. Consider
+   * a REST update method on the message resource collection:      service
+   * Messaging {       rpc UpdateMessage(UpdateMessageRequest) returns (Message)
+   * {         option (google.api.http) = {           patch:
+   * &quot;/v1/messages/{message_id}&quot;           body: &quot;message&quot;
+   * };       }     }     message UpdateMessageRequest {       string message_id
+   * = 1; // mapped to the URL       Message message = 2;   // mapped to the
+   * body     }  The following HTTP JSON to RPC mapping is enabled, where the
+   * representation of the JSON in the request body is determined by protos JSON
+   * encoding:  HTTP | gRPC -----|----- `PATCH /v1/messages/123456 {
+   * &quot;text&quot;: &quot;Hi!&quot; }` | `UpdateMessage(message_id:
+   * &quot;123456&quot; message { text: &quot;Hi!&quot; })`  The special name
+   * `*` can be used in the body mapping to define that every field not bound by
+   * the path template should be mapped to the request body.  This enables the
+   * following alternative definition of the update method:      service
+   * Messaging {       rpc UpdateMessage(Message) returns (Message) { option
+   * (google.api.http) = {           patch:
+   * &quot;/v1/messages/{message_id}&quot;           body: &quot;*&quot; }; } }
+   * message Message {       string message_id = 1;       string text = 2;     }
+   * The following HTTP JSON to RPC mapping is enabled:  HTTP | gRPC -----|-----
+   * `PATCH /v1/messages/123456 { &quot;text&quot;: &quot;Hi!&quot; }` |
```

Comment:
seems like we're losing formatting on a bunch of comments somewhere along the line; seems like it's probably something that would need to be fixed upstream though?